src/components/common/Directors.vue

```diff
@@ -1124,45 +1128,53 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
         director.deliveryAddress = this.inProgressAddress
       }
 
-      if (!Object.values(this.inProgressMailAddress).every(el => el === undefined)) {
-        director.mailingAddress = this.inProgressMailAddress
-      }
+      if (this.isBComp) {
+        if (!Object.values(this.inProgressMailAddress).every(el => el === undefined)) {
+          director.mailingAddress = this.inProgressMailAddress
+        }
 
-      if (this.inheritDeliveryAddress) {
-        director.mailingAddress = director.deliveryAddress
+        if (this.inheritDeliveryAddress) {
+          director.mailingAddress = director.deliveryAddress
+        }
       }
 
       /* COMPARE changes to original director data, for existing directors */
       if (director.actions.indexOf(Actions.APPOINTED) < 0) {
-        const origDirector = this.directorsOriginal.filter(el => el.id === id)[0]
+        const origDirector = this.directorsOriginal.find(director => director.id === id)
 
-        // check whether address has changed
-        if ((JSON.stringify(origDirector.deliveryAddress) !== JSON.stringify(director.deliveryAddress)) ||
```

Comment:
This doesn't always work because order of properties is not guaranteed.

Anyway, "isEqual" is cleaner.